### PR TITLE
[gguf] Ability to specify some fetch headers

### PR DESCRIPTION
**Pull request by julien-c**
cc @ngxson too

### CODE REVIEW

**Comment by coyotte508** [comment]
hmmm there is already a custom `fetch` function, is there a strong need?

File: packages/gguf/src/gguf.ts [hunk 1]
```diff
@@ -64,6 +64,7 @@ class RangeView {
 			 * Custom fetch function to use instead of the default one, for example to use a proxy or edit headers.
 			 */
 			fetch?: typeof fetch;
+			fetchHeaders?: Record<string, string>;
```
Comment (coyotte508) [inline packages/gguf/src/gguf.ts:67]:
```suggestion
			additionalFetchHeaders?: Record<string, string>;
```

**Comment by julien-c** [comment]
> hmmm there is already a custom `fetch` function, is there a strong need?

Well, for safetensors we have a `credentials` param

File: packages/gguf/src/gguf.ts (hunk 1 above)
Comment (julien-c) [reply packages/gguf/src/gguf.ts]: [2f9339c](https://github.com/huggingface/huggingface.js/pull/653/commits/2f9339c5809223dffb391685117a08e7a014e713)

**Review by coyotte508 — APPROVED** [review]

**Comment by ngxson** [comment]
Thanks for tagging me. I have no comment about this PR. But just for my curiosity, is this functionality needed to access gated model? (with a token passed via header, as I remember?)

**Pull request merged**